src/components/UserAuthForm.tsx

```diff
@@ -0,0 +1,39 @@
+"use client"
+
+import React from "react";
+import { signIn } from "next-auth/react";
+import toast from "react-hot-toast";
+import { ButtonVariantEnum } from "@/config/buttonConfig";
+import FunctionButton from "./ui/FunctionButton";
+
+const UserAuthForm = ({ isSignUp } : { isSignUp: boolean }) => {
+    const [isLoading, setIsLoading] = React.useState<boolean>(false);
+
+    const loginWithGoogle = async () => {
+        setIsLoading(true);
+
+        try {
+            await signIn('google');
+        } catch (error) {
+            toast.error('There was an error logging in using Google');
+        } finally {
+            setIsLoading(false);
+        }
+    }
+
+    return (
+        <div className='container flex flex-col justify-front items-center gap-4'>
+            <div className='flex flex-col space-y-2 text-center'>
+                <h1 style={{color: 'black'}}>{
```

Comment:
Just so you know, TailwindCSS (which we're using) will nullify the usual styles that come with h1-6 in HTML (as well as a bunch of other stuff), hence why we have a seperate component called Heading.

Reply:
Ok great, I hope I'll learn more about Tailwind as I go along, but I think I'll dedicate some time to reading over how that is all setup and such.